.devcontainer/devcontainer.json

```diff
@@ -4,25 +4,23 @@
     "name": "iotedgedev",
     "build": {
         "dockerfile": "Dockerfile",
-        "context": "..",
+        "context": ".."
     },
 
-    "runArgs": [
-        "--network",
-        "host"
-    ],
-
-    "postCreateCommand": "pip install -e .",
+    "runArgs": ["--init", "--privileged"],
     "mounts": [
         // Keep command history 
         "source=ostf-bashhistory,target=/commandhistory,type=volume",
-        // Use hosts docker socket
-        "source=/var/run/docker.sock,target=/var/run/docker.sock,type=bind",
+        // Use docker-in-docker socket
+        "source=dind-var-lib-docker,target=/var/lib/docker,type=volume"
```

Comment:
Why have you changed this to be docker in docker? As far as I recall, it's not recommended to use docker in docker, because of the issues it can cause

Reply:
Actually nvm, I've become uneducated and this is fine after reading this article: https://devopscube.com/run-docker-in-docker/#:~:text=To%20run%20docker%20inside%20docker,sock%20as%20a%20volume.&text=Just%20a%20word%20of%20caution,privileges%20over%20your%20docker%20daemon

Reply:
On a second note, for dind, I don't think you need to mount volume at all